internal/health/health.go

```diff
@@ -3,196 +3,24 @@ package health
-	return "/ready", healthResponse()
-}
-
-func healthResponse() func(w http.ResponseWriter, req *http.Request) {
-	return func(w http.ResponseWriter, req *http.Request) {
-		w.WriteHeader(http.StatusOK)
-		fmt.Fprint(w, "OK")
-	}
-}
-
-func runFullControllersCycle(client client.Client, lg *logrus.Entry) func(w http.ResponseWriter, req *http.Request) {
+func handleHealth(etcdURL string) func(w http.ResponseWriter, req *http.Request) {
 	return func(w http.ResponseWriter, req *http.Request) {
-		if err := runAddonsConfigurationControllerCycle(req, client, lg); err != nil {
-			w.WriteHeader(http.StatusInternalServerError)
-			return
-		}
-		if err := runClusterAddonsConfigurationControllerCycle(req, client, lg); err != nil {
-			w.WriteHeader(http.StatusInternalServerError)
-			return
-		}
-
-		w.WriteHeader(http.StatusOK)
-		fmt.Fprint(w, "OK")
-	}
-}
-
-func runAddonsConfigurationControllerCycle(req *http.Request, client client.Client, lg *logrus.Entry) error {
-	probeName := "liveness-probe"
-	probeNamespace := "default"
-
-	addonsConfiguration := &v1alpha1.AddonsConfiguration{
-		ObjectMeta: v1.ObjectMeta{
-			Name:      probeName,
-			Namespace: probeNamespace,
-		},
-		Spec: v1alpha1.AddonsConfigurationSpec{
-			CommonAddonsConfigurationSpec: v1alpha1.CommonAddonsConfigurationSpec{
-				Repositories: []v1alpha1.SpecRepository{{URL: ""}},
-			},
-		},
-	}
-
-	lg.Infof("[liveness-probe] Creating liveness probe addonsConfiguration in %q namespace", probeNamespace)
-	err := client.Create(req.Context(), addonsConfiguration)
-	if err != nil {
-		lg.Errorf("[liveness-probe] Cannot create liveness probe addonsConfiguration: %s", err)
-		return err
-	}
-
-	lg.Info("[liveness-probe] Waiting for liveness probe addonsConfiguration desirable status")
-	err = wait.Poll(1*time.Second, 10*time.Second, func() (done bool, err error) {
-		key := types.NamespacedName{Name: probeName, Namespace: probeNamespace}
-		err = client.Get(req.Context(), key, addonsConfiguration)
-		if apierrors.IsNotFound(err) {
-			lg.Info("[liveness-probe] Liveness probe addonsConfiguration not found")
-			return false, nil
-		}
-		if err != nil {
-			lg.Errorf("[liveness-probe] Cannot get probe addonsConfiguration: %s", err)
-			return false, nil
-		}
-
-		if len(addonsConfiguration.Status.Repositories) != 1 {
-			lg.Info("[liveness-probe] Liveness probe addonsConfiguration repositories status not set")
-			return false, nil
-		}
-
-		status := addonsConfiguration.Status.Repositories[0].Status
-		reason := addonsConfiguration.Status.Repositories[0].Reason
-		if status == v1alpha1.RepositoryStatusFailed {
-			if reason == v1alpha1.RepositoryURLFetchingError {
-				lg.Info("[liveness-probe] Liveness probe addonsConfiguration has achieved the desired status")
-				return true, nil
+		if etcdURL != "" {
+			resp, err := http.Get(fmt.Sprintf("%s%s", etcdURL, "/health"))
+			if err != nil {
+				w.WriteHeader(http.StatusInternalServerError)
+				return
```

Comment:
One of the AC says `helm broker prints error messages to logs in case of etcd connection failure`
should we print/log some err message if http get failed? I am not sure if err has some useful information.

Reply:
We can get an error like: `host not resolved` or `connection timeout`, but I don't want to spam containers' logs with those messages. We could use logger with log levels to print those messages only on develop mode.

If the connection to etcd fails while fetching/saving some data, the error from etcd about connection failure is printed in logs after the timeout.